Update executors-overview.md (#2560)
astronomer/docs: astro/executors-overview.md

The Celery executor is the default for all new Deployments. It uses a group of workers, each of which can run multiple tasks at a time. Astronomer uses [worker autoscaling logic](celery-executor.md#celery-worker-autoscaling-logic) to determine how many workers run on your Deployment at a given time.

> The Celery executor is a good option for most use cases and is the default executor for new Deployments. Specifically, the Celery executor is a good fit for your Deployment if:

Reviewer: ```suggestion
The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:
```

The Celery executor is the default for all new Deployments. It uses a group of workers, each of which can run multiple tasks at a time. Astronomer uses [worker autoscaling logic](celery-executor.md#celery-worker-autoscaling-logic) to determine how many workers run on your Deployment at a given time.

> The Celery executor is a good option for some use cases. Specifically, the Celery executor is a good fit for your Deployment if:

Reviewer: ```suggestion
The Celery executor is a good option for most use cases and is the default executor for new Deployments. Specifically, the Celery executor is a good fit for your Deployment if:
```

Reviewer: I know it's not the right recommendation at the moment, but I don't think product is ready to recommend KE by default yet.

The Celery executor is a good option for some use cases. Specifically, the Celery executor is a good fit for your Deployment if:

> - You're just getting started with Airflow.

Author: Celery has more to configure, and your tasks can crash out - I'd say this is untrue

> - You do not have tasks that require more than 24 hours to execute
>     - Celery workers give their tasks 24 hours to finish so they can restart in the event of a code deployment, though this can be alleviated with [DAG Deploys](deploy-dags.md).

Reviewer: ```suggestion
- You don't have tasks that require more than 24 hours to execute. Celery workers give their tasks only 24 hours to finish so they can restart in the event of a code deployment, although you can get around this with [DAG deploys](deploy-dags.md).
```

Reviewer 2: > although you can get around this with [DAG deploys](deploy-dags.md).

Not sure if this is getting in the weeds, but that only helps for restarts due because of dag deploys. It's still an issue for worker scale-downs.

- You want to use different worker types based on the type of task you're running. See [Configure worker queues](configure-worker-queues.md).

> - You have a high number of short-running tasks
> - Your tasks require the shortest startup latency (often milliseconds)
> - You don't require task or dependency isolation
> - You can configure and fine-tune your Celery workers per deployment

Reviewer: ```suggestion
- You have many short-running tasks.
- Your tasks require the shortest startup latency (often milliseconds).
- You don't require task or dependency isolation.
- You want to fine-tune your workers for each Deployment.
```
The last point applies to KE as well right?

Reviewer 2: I wouldn't put that. If you're talking about celery queues, I'd explicitly call that out (it's a differentiator). i.e `You want to direct tasks to run on different celery worker queues. This is useful if you want to run a subset of tasks with priority for SLA reasons, or if your tasks have varied resource requirements.`

- You have a high number of short-running tasks
- Your tasks require the shortest startup latency (often milliseconds)
- You don't require task or dependency isolation
- You can configure and fine-tune your Celery workers per deployment

> If you regularly experience dependency conflicts or find that some tasks consume the resources of other tasks and cause them to fail, Astronomer recommends implementing worker queues or moving to the Kubernetes executor.

Author: KubeExec doesn't necessarily help with dependency conflicts

Reviewer: Fair enough, I guess you could always use KPO + Celery to achieve the same thing

Reviewer 3: Does it not @fritz-astronomer ?

> The Kubernetes executor runs each task in an individual container. For each task that needs to run, the executor dynamically launches a Kubernetes Pod on your Astro Cluster for the task. When the task completes, the Pod terminates and its resources are released to the cluster. 

Reviewer: ```suggestion
The Kubernetes executor runs each task in an individual container. For each task that needs to run, the executor dynamically launches a Kubernetes Pod on your Astro cluster to run the task. When the task completes, the Pod terminates and its resources are released to the cluster. 
```

On Astro, the Kubernetes infrastructure required to run the Kubernetes executor is built into every Deployment and is managed by Astronomer.

> The Kubernetes executor is a good option for some use cases. Specifically, the Kubernetes executor is a good fit for your Deployment if:

Reviewer 4: I'd avoid repeating "good option for some use cases" in both sections. It seems redundant.

Author: oh - that was actually intentional 😅 I figured peoples' eyes would glaze over and they'd skip some - so I wanted them to relatively mirror each other. I'll let the writers chew on that one, if they deem me worthy of copy-editing @jwitz @lzdanski

Reviewer: Hm I think the main issue here is that we no longer have a good "Default" option for someone just getting started. We recommend Celery in our getting started guides, is that still a sound recommendation? If so, we should say something along those lines here as well.

Reviewer 3: Agreed. I think we have to have a stance on what the default should be, and that's celery right now in our product. A little tricky given limitations but I do think we have to give folks that clarity

Author: It's a way larger conversation but I think KubeExec is *easier* to start with. Celery has config that you need to think about and the possibility to crash out a bunch of tasks and you won't know why. Kube is just *easier* and "I don't need to think about this" IMO

Author: I feel like celery is default because it was first alphabetically 👀

Reviewer: Final call 🏁 Based on some discussions with product I think we need to still recommend Celery as default for the time being, but we should revisit this in the future.

> - You have long-running tasks that may require more than 24 hours to execute.

Reviewer: ```suggestion
- You have long-running tasks that might require more than 24 hours to execute.
```

The Kubernetes executor is a good fit for teams that want fine-grained control over the execution environment of each of their tasks. Specifically, the Kubernetes executor is a good fit for your Deployment if:
The Kubernetes executor is a good option for some use cases. Specifically, the Kubernetes executor is a good fit for your Deployment if:

- You have long-running tasks that require more than 24 hours to execute. The Kubernetes executor ensures that tasks longer than 24 hours are not interrupted when you deploy code.
- You have long-running tasks that may require more than 24 hours to execute.
- Your tasks are compute-intensive or you are processing large volumes of data within the task. Kubernetes executor tasks run separately in a dedicated Pod per task. 
- You have a strong understanding of the CPU and memory that your tasks require and would benefit from being able to allocate and optimize infrastructure resources at the task level.
- Your tasks can tolerate some startup latency (often seconds)
- Your tasks require task or dependency isolation.
- You have had issues running certain tasks reliably with the Celery executor.
  

> The primary limitation with the Kubernetes executor is that each task takes up to 1 minute to start running once scheduled. If you're running short-running tasks and cannot tolerate high latency, Astronomer recommends the Celery executor. To learn more about using the Kubernetes executor, see [Manage the Kubernetes executor](kubernetes-executor.md).

Author: I read "up to a minute" as "it takes minutes to launch my task" - it takes seconds, maybe, to launch a task with KE.

> If only some of your tasks need an isolated execution environment, consider using the [KubernetesPodOperator](kubernetespodoperator.md) with the Celery executor.

Reviewer 4: Not part of the PR but I'd avoid framing this as a catch-all suggestion. KPO is more complex than KE in general.

Reviewer 2: IMO, we should remove it. KPOs and KE serve different use-cases

> If only some of your tasks need an isolated execution environment, consider using the [KubernetesPodOperator](kubernetespodoperator.md) with the Celery executor.

Reviewer: ```suggestion
The Kubernetes executor is primarily used for resource optimization. If you want more control over the environment and dependencies that your tasks run with, consider using the [KubernetesPodOperator](kubernetespodoperator.md) with either the Celery executor or Kubernetes executor.
```

Reviewer: ![Screenshot 2023-07-14 at 12 24 23 PM](https://github.com/astronomer/docs/assets/74574233/6c34a43b-6d35-4af3-8e13-06845bf1b178)
Based on our Learn guide

The Celery executor is a good option for most use cases. Specifically, the Celery executor is a good fit for your Deployment if:

> - You don't have tasks that require more than 24 hours to execute. Celery workers give their tasks only 24 hours to finish so they can restart in the event of a code deploy.

Reviewer: ```suggestion
- You're just getting started with Airflow.
- You don't have tasks that require more than 24 hours to execute. Celery workers give their tasks only 24 hours to finish so they can restart in the event of a code deploy.
```
